state/indexer/block/kv/kv_test.go

```diff
@@ -0,0 +1,141 @@
+package kv_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/pubsub/query"
+	blockidxkv "github.com/tendermint/tendermint/state/indexer/block/kv"
+	"github.com/tendermint/tendermint/types"
+	db "github.com/tendermint/tm-db"
+)
+
+func TestBlockIndexer(t *testing.T) {
+	store := db.NewPrefixDB(db.NewMemDB(), []byte("block_events"))
+	indexer := blockidxkv.New(store)
+
+	require.NoError(t, indexer.Index(types.EventDataNewBlockHeader{
+		Header: types.Header{Height: 1},
+		ResultBeginBlock: abci.ResponseBeginBlock{
+			Events: []abci.Event{
+				{
+					Type: "begin_event",
+					Attributes: []abci.EventAttribute{
+						{
+							Key:   []byte("proposer"),
+							Value: []byte("FCAA001"),
+							Index: true,
+						},
+					},
+				},
+			},
+		},
+		ResultEndBlock: abci.ResponseEndBlock{
+			Events: []abci.Event{
+				{
+					Type: "end_event",
+					Attributes: []abci.EventAttribute{
+						{
+							Key:   []byte("foo"),
+							Value: []byte("100"),
+							Index: true,
+						},
+					},
+				},
+			},
+		},
+	}))
+
+	for i := 2; i < 12; i++ {
```

Comment:
why 2/12?

Reply:
We index height 1 explicitly before this loop and 12 is completely arbitrary, enough to make sure we index a few blocks :)